src/IFileDownloader.ts

```diff
@@ -2,7 +2,14 @@
 // Licensed under the MIT License.
 
 import { ExtensionContext, CancellationToken, Uri } from "vscode";
-import { FileDownloadSettings } from "./FileDownloader";
+
+export interface FileDownloadSettings {
+    timeoutInMs?: number;
+    retries?: number;
+    retryDelayInMs?: number;
+    shouldUnzip?: boolean;
+    headers?: Record<string, string | number | boolean> | undefined;
```

Comment:
💡 Thoughts\Idea: If it's possible can we please add little reference and for `header and the Record` type usage in the context. Code comment or relevance in the PR will be great for archive reasons. Thanks.

Reply:
Added some comments to the interface